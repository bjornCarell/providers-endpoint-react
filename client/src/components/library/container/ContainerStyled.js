import styled from 'styled-components';

export const Container = styled.div`
  margin: 0 auto;
  padding-bottom: ${props => {
    if (props.fullVertical) return 0;
    if (props.small) return '1.5rem';
    return '2.5rem';
  }};
  padding-left: ${props => {
    if (props.full || props.noPaddingLeft) return 0;
    return 'calc((100vw - 96rem) / 2)';
  }};
  padding-right: ${props => {
    if (props.full || props.noPaddingRight) return 0;
    return 'calc((100vw - 96rem) / 2)';
  }};
  padding-top: ${props => {
    if (props.fullVertical) return 0;
    if (props.small) return '1.5rem';
    return '2.5rem';
  }};
`;
