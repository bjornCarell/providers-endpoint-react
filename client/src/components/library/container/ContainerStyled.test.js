import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Container } from './ContainerStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(
    <Container>
      <div>Providers endpoint</div>
    </Container>,
    div
  );
});

// SNAPSHOTS
it('renders correctly with fullVertical prop', () => {
  const tree = renderer.create(<Container fullVertical />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('padding-bottom', '0');
  expect(tree).toHaveStyleRule('padding-top', '0');
});

it('renders correctly with noPaddingLeft prop', () => {
  const tree = renderer.create(<Container noPaddingLeft />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('padding-left', '0');
});

it('renders correctly with noPaddingRight prop', () => {
  const tree = renderer.create(<Container noPaddingRight />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('padding-right', '0');
});
