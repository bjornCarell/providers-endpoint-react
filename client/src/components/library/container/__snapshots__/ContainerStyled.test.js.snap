// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with fullVertical prop 1`] = `
.c0 {
  margin: 0 auto;
  padding-bottom: 0;
  padding-left: calc((100vw - 96rem) / 2);
  padding-right: calc((100vw - 96rem) / 2);
  padding-top: 0;
}

<div
  className="c0"
/>
`;

exports[`renders correctly with noPaddingLeft prop 1`] = `
.c0 {
  margin: 0 auto;
  padding-bottom: 2.5rem;
  padding-left: 0;
  padding-right: calc((100vw - 96rem) / 2);
  padding-top: 2.5rem;
}

<div
  className="c0"
/>
`;

exports[`renders correctly with noPaddingRight prop 1`] = `
.c0 {
  margin: 0 auto;
  padding-bottom: 2.5rem;
  padding-left: calc((100vw - 96rem) / 2);
  padding-right: 0;
  padding-top: 2.5rem;
}

<div
  className="c0"
/>
`;
